client/exec.go

```
@@ -252,22 +271,33 @@ func matchRegexp(cmd string, regexpList []*regexp.Regexp) bool {
 }
 
 type CapacityBuffer struct {
-	bytes.Buffer
-	capacity int
+	data        []byte
+	capacity    int
+	hasOverflow bool
+}
+
+func (b *CapacityBuffer) HasOverflow() bool {
+	return b.hasOverflow
+}
+
+func (b *CapacityBuffer) GetOverflowMessage() string {
+	return fmt.Sprintf("maximum send_back_limit of %d bytes exceeded", b.capacity)
 }
 
 func (b *CapacityBuffer) Write(p []byte) (n int, err error) {
-	freeCapacity := b.capacity - b.Len()
+	freeCapacity := b.capacity - len(b.data)
 
 	// do not write to buffer if no space left
-	if freeCapacity <= 0 {
-		return len(p), nil // pretend a successful write
-	}
-
-	// write to buffer only a part if exceeds the capacity
 	if len(p) > freeCapacity {
-		return b.Buffer.Write(p[:freeCapacity])
+		b.hasOverflow = true
+		return 0, errors.New(b.GetOverflowMessage())
```

Are you sure returning an error here is a good idea? Previous implementation would just silently drop the overflow data and pretend everything is ok for the exec package.
Have you tested what the `exec` package does when error happens writing to stderr or stdout? Does it stop the execution of the command or continues? Does it report the error somewhere?

---

it was the requirement of the task - return error is buffer is overflown 
I've tested it the exec returns "broken pipe error" so I have to catch it in a different way (through the flag)
yes it does stop the execution with an stderr